alacritty/src/display/content.rs

```diff
@@ -350,11 +350,13 @@ impl RenderableCell {
     /// using the named input color, rather than checking the RGB of the background after its color
     /// is computed.
     #[inline]
-    fn compute_bg_alpha(bg: Color) -> f32 {
+    fn compute_bg_alpha(content: &mut RenderableContent<'_>, bg: Color) -> f32 {
         if bg == Color::Named(NamedColor::Background) {
             0.
-        } else {
+        } else if content.config.ui_config.colors.opaque_background_colors {
             1.
+        } else {
+            content.config.ui_config.background_opacity()
```

Comment:
In theory we could take a reference to ui_config as argument to this method, just to keep things minimal and reusable in theory. But it doesn't matter *that* much.